### PR TITLE
RELATED: RAIL-3597 Handle multiple drills on attribute

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-3597

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivanmjartan** [comment]
ok to test

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts [hunk 1]
```diff
@@ -16,3 +18,30 @@ export function getDrillDownAttributeTitle(drill: IDrillDownDefinition, drillEve
...
+ */
+export function filterDrillFromAttributeByPriority(
+    drillDefinitions: DashboardDrillDefinition[],
+): DashboardDrillDefinition[] {
+    const nonImplicitDrillDown = drillDefinitions.filter(
+        (dd) => !isDrillDownDefinition(dd) && !isDrillToLegacyDashboard(dd),
+    );
+
+    const nonImplicitDrillDownIds = nonImplicitDrillDown.map((d) => {
+        if (!isDrillToLegacyDashboard(d)) {
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts:35]: Is this type guard necessary? (as you already use it in the previous filter fn)

Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts:35]: Will this if ever be false? These will be filtered out by the filter on line 30 no?

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts [hunk 2]
```diff
@@ -16,3 +18,30 @@ export function getDrillDownAttributeTitle(drill: IDrillDownDefinition, drillEve
...
+
+/**
+ * Implicit drillDown has lower priority, so needs to be removed when other drill config exists for the same attribute
+ *
+ * @internal
+ */
+export function filterDrillFromAttributeByPriority(
+    drillDefinitions: DashboardDrillDefinition[],
+): DashboardDrillDefinition[] {
+    const nonImplicitDrillDown = drillDefinitions.filter(
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts:30]: Nitpick: Is the naming correct? These are not only drillDowns, or am I missing something? Maybe drillDefinitionsWithoutDrillDown or possibleConflictingDrillDefinitions ? I have no idea about better naming... or maybe just "drills" + comment about the reason of filtering them ? :/

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts:35]: Discussed offline, it is because of typescript inference

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4733/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6366/ : SUCCESS in 20m 41s

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts]: Done

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts (hunk 2 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts]: renamed and use just one filter

File: libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/drill/utils/drillDownUtils.ts]: use just one filter

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1337/ : SUCCESS in 1m 03s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1613/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/391/ : SUCCESS in 11m 33s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6368/ : SUCCESS in 17m 04s

**Pull request merged**